### PR TITLE
[Feat] Adding respawning tiers

### LINKED ISSUE #6: [Bug] Error thrown on 0.2.0
Stacktrace:
```
Could not pass event EntityDamageByEntityEvent to AlteredRespawn v0.1.1
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_19_R3.event.CraftEventFactory.callEvent(CraftEventFactory.java:263) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at org.bukkit.craftbukkit.v1_19_R3.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:1071) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at org.bukkit.craftbukkit.v1_19_R3.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:938) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at org.bukkit.craftbukkit.v1_19_R3.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:887) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at org.bukkit.craftbukkit.v1_19_R3.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:1105) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityLiving.damageEntity0(EntityLiving.java:2045) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.player.EntityHuman.damageEntity0(EntityHuman.java:1028) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityLiving.a(EntityLiving.java:1349) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.player.EntityHuman.a(EntityHuman.java:935) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.level.EntityPlayer.a(EntityPlayer.java:979) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityInsentient.z(EntityInsentient.java:1615) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.monster.EntityZombie.z(EntityZombie.java:347) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.ai.goal.PathfinderGoalMeleeAttack.a(SourceFile:152) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.ai.goal.PathfinderGoalMeleeAttack.e(SourceFile:144) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.ai.goal.PathfinderGoalZombieAttack.e(SourceFile:28) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.ai.goal.PathfinderGoalWrapped.e(SourceFile:65) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.ai.goal.PathfinderGoalSelector.a(SourceFile:120) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.ai.goal.PathfinderGoalSelector.a(SourceFile:111) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityInsentient.eY(EntityInsentient.java:887) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityLiving.b_(EntityLiving.java:3121) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityInsentient.b_(EntityInsentient.java:657) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.monster.EntityMonster.b_(SourceFile:44) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.monster.EntityZombie.b_(EntityZombie.java:264) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityLiving.l(EntityLiving.java:2852) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.EntityInsentient.l(EntityInsentient.java:419) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.entity.monster.EntityZombie.l(EntityZombie.java:234) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.level.WorldServer.a(WorldServer.java:847) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.level.World.a(World.java:689) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.level.WorldServer.lambda$9(WorldServer.java:425) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.world.level.entity.EntityTickList.a(SourceFile:54) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.level.WorldServer.a(WorldServer.java:405) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1304) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:388) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1200) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1017) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: java.lang.IllegalArgumentException: Health must be between 0 and 6.0(20.0)
	at org.bukkit.craftbukkit.v1_19_R3.entity.CraftLivingEntity.setHealth(CraftLivingEntity.java:117) ~[spigot-1.19.4-R0.1-SNAPSHOT.jar:3763-Spigot-7d7b241-5a5e43e]
	at world.anhgelus.msmp.alteredrespawn.api.Stack$deathEvent$1.invoke(Stack.kt:47) ~[?:?]
	at world.anhgelus.msmp.alteredrespawn.api.Stack$deathEvent$1.invoke(Stack.kt:44) ~[?:?]
	at world.anhgelus.msmp.msmpcore.player.MPlayer.died(MPlayer.kt:112) ~[?:?]
	at world.anhgelus.msmp.alteredrespawn.api.Stack.deathEvent(Stack.kt:44) ~[?:?]
	at world.anhgelus.msmp.alteredrespawn.listener.RegisterListener.onDamage(RegisterListener.kt:14) ~[?:?]
	at jdk.internal.reflect.GeneratedMethodAccessor9.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:?]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-api-1.19.4-R0.1-SNAPSHOT.jar:?]
	... 40 more
```

Seems like to be linked with a previous [bug](https://github.com/msmp-mc/msmp-core/issues/4) in MSMPCore

### LINKED ISSUE #7: [Bug] Incompatibility between absorbtion hearts and gapple / notch apple
When we eat a golden apple (or a notch apple) after beeing revived, our hearts will do this:

1. First time, regive you all your absorbtion hearts
2. Other times, never give your absorbtion hearts

### LINKED ISSUE #8: [Bug] We can take the helmet of the armor stands
We can take the helmet of the armor stands.

### LINKED ISSUE #9: [Feat] Disable "high flying" for spectator
When a spectator decide to go to a high flying (higher than the walls), he can try to find the exit.

To fix it, we must rollback the user after beeing at this state.

### PR DESCRIPTION
**Pull request by Akik4**

### CODE REVIEW

File: src/main/kotlin/world/anhgelus/msmp/alteredrespawn/api/Stack.kt [hunk 1]
```diff
@@ -101,13 +108,47 @@ object Stack {
                         respawned.sendTitle("respawned !", null, 0, 20, 0)
                         mplayer.toDefaultCondition()
                         respawned.world.strikeLightningEffect(event.rightClicked.location)
+                        respawned.addPotionEffect(PotionEffectType.BLINDNESS.createEffect(80, 5))
+                        respawned.addPotionEffect(PotionEffectType.SLOW.createEffect(160, 10))
+                        respawned.addPotionEffect(PotionEffectType.WEAKNESS.createEffect(160, 5))
+                        respawned.addPotionEffect(PotionEffectType.CONFUSION.createEffect(160, 1))
+                        this.setTier(respawned, helmet.color.asARGB())
```
Comment (anhgelus) [inline src/main/kotlin/world/anhgelus/msmp/alteredrespawn/api/Stack.kt:117]: I think we can create a config to modify it easily.

File: src/main/kotlin/world/anhgelus/msmp/alteredrespawn/api/Stack.kt [hunk 2]
```diff
@@ -101,13 +108,47 @@ object Stack {
...
+            }
+            DyeColor.BLUE.color.asARGB() -> {
+                player.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.baseValue = 8.0
+                player.getAttribute(Attribute.GENERIC_MOVEMENT_SPEED)?.baseValue = 0.4
+                player.addPotionEffect(PotionEffectType.ABSORPTION.createEffect(Int.MAX_VALUE, 15))
+            }
+            else-> {
+                player.getAttribute(Attribute.GENERIC_MAX_HEALTH)?.baseValue = 6.0
+                player.getAttribute(Attribute.GENERIC_MOVEMENT_SPEED)?.baseValue = 0.5
+                player.addPotionEffect(PotionEffectType.ABSORPTION.createEffect(Int.MAX_VALUE, 20))
```
Comment (anhgelus) [inline src/main/kotlin/world/anhgelus/msmp/alteredrespawn/api/Stack.kt]: Same for everything here.

**Review by anhgelus — CHANGES_REQUESTED** [review]

**Comment by anhgelus** [comment]
And don't forget to set the version to `0.2.0`

**Comment by anhgelus** [comment]
Known bug: #6

File: src/main/kotlin/world/anhgelus/msmp/alteredrespawn/api/Stack.kt (hunk 1 above)
Comment (Akik4) [reply src/main/kotlin/world/anhgelus/msmp/alteredrespawn/api/Stack.kt:117]: For ? That when player died

**Review by anhgelus — APPROVED** [review]

**Pull request merged**